app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,79 @@
+<div class="m-2 border-2 border-black p-2 items-center rounded">
+	<div class="w-full my-4 flex justify-between items-center">
+	<h2 class="m-4 text-indigo-black font-bold"><%= @recipe.name %></h2>
+	<% if !current_user.nil? && current_user.id == @recipe.user_id %>
+		<%= form_with model:@recipe, url:recipe_path(@recipe.id), html:{method: :put} do |form| %>
+		    <div class="p-2 border-black border-1 rounded">
+		      <%= form.label :public, class:"text-xl ml-5 flex-1 p-4" %>
+		      <%# <%= form.check_box :public, class: "mr-5 flex-1 border-black border-4 p-4 rounded", onchange: "this.form.submit()" %>
+					<input class="form-check-input" type="checkbox" id="toggle-public" <%= @recipe.public ? "checked" : "" %>>
+		    </div>
+		<% end %>
+	<% end %>
+	</div>
+	<div class="text-indigo-bold w-full m-2 flex justify-start items-center">
+	<h3 class="font-semibold">Preparation time: </h3><h6 class="ml-2"> <%= @recipe.preparation_time %></h6>
+	</div>
+	<div class="text-indigo-black w-full m-2 flex justify-start items-center">
+	<h3 class="font-semibold">Cooking time: </h3><h6 class="ml-2"> <%= @recipe.cooking_time %></h6>
+	</div>
+	<div class="w-full m-2 flex justify-start items-center">
+	<h3 class="text-indigo-black font-semibold">Description: </h3><p class="ml-2 w-full"> <%= @recipe.description %></p>
+	</div>
+	<% if !current_user.nil? && current_user.id == @recipe.user_id %>
+		<div class="w-full my-4 flex justify-between items-center">
+			<%= link_to "Generate shopping list", recipe_new_shopping_list_path(@recipe.id), class: "bg-red-500 px-4 py-2 font-semibold text-white inline-flex items-center space-x-2 rounded"%>
```

Comment:
- [ ] onclick of this generate shopping should take me to shopping list page, but now it throws an error if the recipe has ingredients, kindly fix this
![Screenshot from 2023-02-18 17-25-45](https://user-images.githubusercontent.com/66097371/219876901-d99eb4f0-0619-44b5-b0d8-de5772ae02ab.png)